### PR TITLE
ref: document `dvc exp clean`

### PR DESCRIPTION
**Pull request by pmrowla**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Docs PR for https://github.com/iterative/dvc/pull/9090

### CODE REVIEW

File: content/docs/command-reference/exp/clean.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# exp clean
+
+Cleanup internal DVC experiment temporary files.
```
Comment (shcheklein) [inline content/docs/command-reference/exp/clean.md:3]: DVC experiment -> [experiments] (like we do in the other commands)

minor: can we drop internal? just keep temporary - I think it's more or less clear that they are internal then?

File: content/docs/command-reference/exp/clean.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Synopsis
+
+```usage
+usage: dvc exp clean [-h] [-q | -v]
+```
+
+## Description
+
+Runs housekeeping tasks within the DVC repository, such as removing outdated
```
Comment (shcheklein) [inline content/docs/command-reference/exp/clean.md:13]: minor: I think `DVC repository` can be wrapped into `<abbr>` (I would check something similar in docs)

**Review by shcheklein — COMMENTED** [review]
some minor comments,

@yathomasi do we need anything to do to enable syntax highlight for the new command?

**Comment by yathomasi** [comment]
> do we need anything to do to enable syntax highlight for the new command?

https://github.com/iterative/gatsby-theme-iterative/blob/main/packages/gatsby-theme-iterative/config/prismjs/dvc-commands.js

We need to add the command to this list on the theme.

**Comment by shcheklein** [comment]
Added https://github.com/iterative/gatsby-theme-iterative/pull/182. We'll need to bump the theme later.

**Review by omesser — APPROVED** [review]

**Review by shcheklein — APPROVED** [review]

File: content/docs/command-reference/exp/index.md [hunk 3]
```diff
@@ -37,6 +38,7 @@ positional arguments:
     gc                  Garbage collect unneeded experiments.
     push                Push a local experiment to a Git remote.
     pull                Pull an experiment from a Git remote.
+    clean               Cleanup dvc exp internal tempfiles.
```
Comment (shcheklein) [inline content/docs/command-reference/exp/index.md:41]: a possible followup on DVC itself: 

`dvc exp` -> `DVC experiments` (as in the `dvc exp init`)

or actually simplify both to just `experiments` (it's clear that it's DVC I think), and in some other commands we already do this `Remove local experiments.`, we don't do `Remove DVC experiments`. 

I can send a PR if that makes sense @pmrowla , let me know.

Comment (pmrowla) [reply content/docs/command-reference/exp/index.md:41]: Yeah that's fine with me @shcheklein

Comment (shcheklein) [reply content/docs/command-reference/exp/index.md:41]: https://github.com/iterative/dvc/pull/9149

**Pull request merged**